### PR TITLE
package ramen alerts

### PR DESCRIPTION
**Pull request by rakeshgm**
adds `VolumeSynchronizationDelay` alerts. 
[BZ: 2219843](https://bugzilla.redhat.com/show_bug.cgi?id=2219843)

### CODE REVIEW

**Comment by rakeshgm** [comment]
I have tested in minikube cluster. I need check once more in Openshiftt. will open the PR once i am done with testing.

File: config/prometheus/alerts.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      rules:
+      - record: ramen_sync_duration_seconds
+        expr: sum by (obj_name, obj_namespace, obj_type, job, policyname)(time() - (ramen_last_sync_timestamp_seconds > 0))
+      - record: ramen_rpo_difference
+        expr: ramen_sync_duration_seconds{job="ramen-hub-operator-metrics-service"} / on(policyname, job) group_left() (ramen_policy_schedule_interval_seconds{job="ramen-hub-operator-metrics-service"})
+      - alert: VolumeSynchronizationDelay
+        expr: ramen_rpo_difference >= 3
+        for: 5s
+        labels:
+          cluster: "{{ $labels.cluster }}"
```
Comment (GowthamShanmugam) [inline config/prometheus/alerts.yaml:21]:
```suggestion
        
```
OCP Thanos don't have this label

File: config/prometheus/alerts.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          severity: critical
+        annotations:
+          description: "Syncing of volumes (DRPC: {{ $labels.obj_name }}, Namespace: {{ $labels.obj_namespace }}) is taking more than thrice the scheduled snapshot interval. This may cause data loss and a backlog of replication requests. To get around the delay, follow the instructions provided in the documentation."
+          alert_type: "DisasterRecovery"
+      - alert: VolumeSynchronizationDelay
+        expr: ramen_rpo_difference > 2 and ramen_rpo_difference < 3
+        for: 5s
+        labels:
+          cluster: "{{ $labels.cluster }}"
```
Comment (GowthamShanmugam) [inline config/prometheus/alerts.yaml:30]:
```suggestion
          
```
OCP Thanos don't have this label

**Comment by GowthamShanmugam** [comment]
I tested on Aws cluster

File: config/prometheus/alerts.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    prometheus: k8s
+    role: alert-rules
+  name: alerting-rules
+  namespace: system
+spec: 
+  groups:
+    - name: alerts
+      rules:
+      - record: ramen_sync_duration_seconds
+        expr: sum by (obj_name, obj_namespace, obj_type, job, policyname)(time() - (ramen_last_sync_timestamp_seconds > 0))
```
[the remark this answers is not in the document]
Comment (rakeshgm) [reply config/prometheus/alerts.yaml]: changes added

File: config/prometheus/alerts.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      - record: ramen_rpo_difference
+        expr: ramen_sync_duration_seconds{job="ramen-hub-operator-metrics-service"} / on(policyname, job) group_left() (ramen_policy_schedule_interval_seconds{job="ramen-hub-operator-metrics-service"})
```
[the remark this answers is not in the document]
Comment (rakeshgm) [reply config/prometheus/alerts.yaml]: changes added

File: config/prometheus/alerts.yaml (hunk 1 above)
Comment (rakeshgm) [reply config/prometheus/alerts.yaml]: removed the label

File: config/prometheus/alerts.yaml (hunk 2 above)
Comment (rakeshgm) [reply config/prometheus/alerts.yaml]: removed the label

**Review by GowthamShanmugam — APPROVED** [review]

File: config/prometheus/alerts.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  name: alerting-rules
+  namespace: system
+spec: 
+  groups:
+    - name: alerts
+      rules:
+      - record: ramen_sync_duration_seconds
+        expr: (time() - (ramen_last_sync_timestamp_seconds{job='ramen-hub-operator-metrics-service'} > 0))
+      - record: ramen_rpo_difference
+        expr: ramen_sync_duration_seconds / on(policyname) group_left() (ramen_policy_schedule_interval_seconds{job="ramen-hub-operator-metrics-service"})
```
Comment (umangachapagain) [inline config/prometheus/alerts.yaml:16]: These are recording rules and not alerting rules. So, it's better to group these separately as "recording rules".

**Review by umangachapagain — APPROVED** [review]

**Review by ShyamsundarR — APPROVED** [review]

**Comment by agarwal-mudit** [comment]
@rakeshgm pls backport this to red-hat-storae: release-4.14

**Pull request merged**